igc.cc

```diff
@@ -400,12 +437,74 @@ void IgcFormat::read()
       }
       break;
 
+    case rec_fix_defn:
+      /*
+         The first three characters define the number of extensions present.
+         We don't particularly care about that. After that, every group of seven
+         bytes is 4 digits followed by three letters, specifying start end end
+         bytes of each extension, and the kind of extension (always three chars)
+      */
+    
+        // QString ibuf_q = QString::fromLocal8Bit(ibuf);
+        // QStringList ext_data = ibuf_q.split(QRegularExpression ("[A-Z]+"));
+
+      for (unsigned i = 3; i< strlen(ibuf); i+=7){
+        s = ibuf;
+        s2 = s.substr(i, 7);
+        sscanf(s2.c_str(), "%2i%2i%3s", &begin, &end, ext_record_type);
+        switch (hash2stringint(ext_record_type)) {
+          case hash2stringint("ENL"):
```

Comment:
QStringView looks good from here. QStringRef is going away in Qt6, I just learned. It's an ugly hack because I was trying to switch on a const string anyway.

I'll need to look up how exactly I can cast a QStringView to an int, though. That is not immediately understandable.